src/main/java/com/vinylteam/vinyl/security/impl/DefaultSecurityService.java

```diff
@@ -66,18 +72,20 @@ public User createUserWithHashedPassword(String email, char[] password) {
         user.setIterations(iterations);
         user.setRole(Role.USER);
         user.setStatus(false);
+        logger.debug("Resulting user is {'user':{}}", user);
         return user;
     }
 
     @Override
     public boolean checkPasswordAgainstUserPassword(User user, char[] password) {
-        if (user == null) {
-            return false;
-        } else {
-            boolean isSameHash;
-            isSameHash = (user.getPassword().equals(
-                    hashPassword(password, Base64.getDecoder().decode(user.getSalt()), user.getIterations())));
-            return isSameHash;
+        boolean isSameHash = false;
+        if (user != null) {
+            isSameHash = (user.getPassword().equals(hashPassword(password,
+                    Base64.getDecoder().decode(user.getSalt()), user.getIterations())));
+            logger.debug("Compared hash of passed password against user's hashed password");
         }
+        logger.debug("Result of comparing hash of passed password against user's hashed password" +
+                "if user isn't null is {'isSameHash': {}, 'user':{}}", isSameHash, user);
```

Comment:
Please make it shorter and use one row for log. It is look like documentation(too much info)